assets/eip-4910/eip-4910-reference-implementation/LICENSE

```diff
@@ -0,0 +1,223 @@
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+                        The MIT License (MIT)
+
+Copyright (c) 2018 Truffle
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
```

Comment:
Everything added to the EIP repository needs to be CC0 licensed.  If this isn't your own code, then it means you can't license it as CC0 which means it cannot be added unfortunately.  Note: Reference implementation is not strictly required for EIPs.

Reply:
@MicahZoltu so how do you do reference implementations when you inherit from contracts like open zeppelin? That does not seem to make a lot of sense if you want to make it easy for others to try it out. And I think that a reference implementation should always accompany a standard. They do not necessarily have to have the same licensing, as long as it is not restrictive e.g. Apache 2.0 for reference implementations should be ok. And testing suites like truffle should also be ok as long as it is a permissive license.

I can certainly remove the reference implementation if that is the only way to get the EIP merged. Please, let me know.

Reply:
You don't need to include the code for the dependencies.  You can just write something like:
```
interface IEIP1234 extends IEIP20 {
	...
}
```
You don't need to actually include `IEIP20`, the reader can go find any implementation of the standard online that they want to use (which may include Open Zeppelin or something else).

The purpose of the reference implementation isn't to provide users with a tutorial or a getting started kit.  It is to provide some example code that shows implementing this standard is *possible* and what that might look like.  Generally speaking, the reference implementation doesn't need to be production ready either, which means you don't need to do things like reentrancy protection and whatnot.

I think the key takeaway here is that an EIP is *NOT* meant to be a place where people go to learn how to use a standard.  People should write blogs, tutorials, sample repositories, etc. for all of that.  The EIP should contain the specification and minimal supplemental material for that.  Everything else goes elsewhere.

Reply:
removed reference implementation